docs/ref/models/fields.txt

```diff
@@ -479,7 +485,9 @@ should be the value of the field they reference (``pk`` unless
 
 The default value is used when new model instances are created and a value
 isn't provided for the field. When the field is a primary key, the default is
-also used when the field is set to ``None``.
+also used when the field is set to ``None``. When the field is string-based
+and a default is not provided, Django automatically uses an empty string
+(see :attr:`~Field.null`).
 
 The default value can also be set at the database level with
 :attr:`Field.db_default`.
```

Comment:
If we put anything here I think maybe we should have something like
```
See :meth:`Field.get_default` for how the default value is determined when
``default`` is not set.
```
and further down under the `descriptor_class` docs
```

    .. method:: get_default()

        Returns the default value of the field. This is :attr:`~Field.default`
        or :attr:`~Field.db_default`, with :attr:`~Field.default` taking
        precedence over :attr:`Field.db_default` if both are set. If neither are
        set, then returns an empty string if the field allows it, otherwise
        returns ``None``.
```

But I'm not sure if a comment about `blank` might be enough

Reply:
Yes I agree. That's certainly better than my addition in ``Field.default``. I wasn't able to find the docs for ``Field.get_default`` previously.

Reply:
I'm not sure who should add the documentation for ``Field.get_default()``. Would you like to handle it, or should I include it in the Pull Request?

Reply:
I've decided to remove `Field.get_default()` from the pull request for now as I am not 100% sure whether the clarity we gain is worth adding this to the public API :thinking: 
But moving forward with the other changes :+1: